ui/src/utils/search-sort-incarnations.test.ts

```diff
@@ -88,6 +91,24 @@ describe('searchSortIncarnations', () => {
       'dir/5'
     ].reverse())
   })
+  it('should sort by template version', () => {
```

Comment:
could we add some tests that also verify behavior when the version is not actually a semantic version? But a commit ID or a branch name?